src/main/scala/org/constellation/genesis/GenesisObservationWriter.scala

```diff
@@ -0,0 +1,69 @@
+package org.constellation.genesis
+
+import better.files.File
+import cats.data.EitherT
+import cats.effect.{Concurrent, ContextShift, Sync}
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+import org.constellation.domain.schema.Id
+import org.constellation.primitives.Schema.GenesisObservation
+import org.constellation.serializer.KryoSerializer
+import org.constellation.storage.external.CloudStorage
+import org.constellation.{ConfigUtil, DAO}
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+
+class GenesisObservationWriter[F[_]: Concurrent: Logger](
+  cloudStorage: CloudStorage[F],
+  dao: DAO,
+  executionContext: ContextShift[F],
+  fileOperationContext: ExecutionContext
+) {
+
+  def write(genesisObservation: GenesisObservation): EitherT[F, GenesisObservationWriterException, Unit] =
+    for {
+      _ <- EitherT(writeToDisk(genesisObservation))
+      _ <- EitherT.liftF(Logger[F].info("Genesis observation saved on disk"))
+
+      result <- sendToCloudIfRequired(genesisObservation)
+      _ <- EitherT.liftF(Logger[F].info(s"Genesis observation files saved on cloud : ${result.size}"))
+    } yield ()
+
+  private def writeToDisk(genesis: GenesisObservation): F[Either[GenesisObservationWriterException, File]] =
+    executionContext.evalOn(fileOperationContext) {
+      Sync[F].delay {
+        Try(getFilePathToGenesisObservation.writeByteArray(KryoSerializer.serializeAnyRef(genesis))) match {
```

Comment:
How about using `Resource` for files? https://typelevel.org/cats-effect/datatypes/resource.html

Reply:
I think that would be also good.
But here it is not a pure java `File` class but also a wrapper to Java NIO, https://github.com/pathikrit/better-files